spark/v3.5/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckViews.scala

```diff
@@ -36,6 +38,13 @@ object CheckViews extends (LogicalPlan => Unit) {
         verifyColumnCount(ident, columnAliases, query)
         SchemaUtils.checkColumnNameDuplication(query.schema.fieldNames, SQLConf.get.resolver)
 
+      case UnsetViewProperties(ResolvedV2View(catalog, ident), propertyKeys, ifExists) =>
```

Comment:
Check rules are intended to ensure that SQL is correct and valid, not to check the execution. If you have this check here, it would fail when you run `EXPLAIN ALTER VIEW ...` because this is doing execution work (checking for the property) in the planner. Another way to think about this is that this isn't an analysis failure, it is a runtime failure due to data. The SQL is perfectly valid.

This should either be done in `AlterV2ViewExec` or not at all. I would lean toward not doing this check at all. Shouldn't this be idempotent? Is there value in failing to remove a property if it isn't there? I don't see value so I'd ignore this and let the catalog throw an exception if it chooses to.

Reply:
Okay, I missed the `ifExists` part. looks like this behavior is required by Spark.

In that case, this should be moved to the Exec plan.